descriptors/ModuleDescriptor-template.json

```diff
@@ -222,26 +222,10 @@
       "id" : "login",
       "version" : "6.0 7.0"
     },
-    {
-      "id" : "service-points",
-      "version" : "2.1 3.0"
-    },
-    {
-      "id" : "service-points-users",
-      "version" : "1.0"
-    },
-    {
-      "id": "password-validator",
-      "version": "1.0"
-    },
     {
       "id": "authtoken",
       "version": "2.0"
     },
-    {
-      "id": "notify",
-      "version": "2.0"
-    },
```

Comment:
When an optional dependency is missing but is required for some API mod-users-bl gets a 404 HTTP response from Okapi. In this case mod-users-bl should return a good error message that indicates the cause of this error, and this should be tested by unit tests. Please add these error messages and unit tests.

Reply:
> When an optional dependency is missing but is required for some API mod-users-bl gets a 404 HTTP response from Okapi. In this case mod-users-bl should return a good error message that indicates the cause of this error 

I disagree that the endpoint should produce an error.

What is the point of having an endpoint that cannot succeed (with the same configuration)?

If an endpoint cannot achieve it's aim without the presence of a dependency, then it isn't an optional dependency.

> this should be tested by unit tests. Please add these error messages and unit tests.

I agree it would be good to demonstrate the behaviour of the module when dependencies are missing in tests.

Reply:
We define dependencies on the module level, not on the API level.

Given that
* module A has API /a/x that always works, it doesn't need module B, and
* module A has API /a/y that needs module B and and returns a failure message ("module B missing") if missing. 

In this example module A can declare module B as optional dependency because an institution that only needs API /a/x and neither needs API /a/y nor module B should be able to install module A but not module B.

Reply:
> In this example module A can declare module B as optional dependency because an institution that only needs API /a/x and neither needs API /a/y nor module B should be able to install module A but not module B.

How does that fit with your statement above that we define dependencies on the module level?

Given that we define dependencies on the whole interface, how does it make sense for some the modules API to be completely undermined when a dependency isn't present?

Reply:
> module A has API /a/y that needs module B and and returns a failure message ("module B missing") if missing.

If this is acceptable, we could make all dependencies optional and fail every API at runtime.